### PR TITLE
Cancel prev CI runs on new pushes

### PR DESCRIPTION
**Pull request by robhudson**
See:
https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/control-the-concurrency-of-workflows-and-jobs

### CODE REVIEW

File: .github/workflows/pull_request_tests.yml [hunk 1]
```diff
@@ -5,6 +5,10 @@ on:
       - "main"
   pull_request:
 
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
```
Comment (pmac) [inline .github/workflows/pull_request_tests.yml:9]: This looks good. Can we also do this (or instead) with branch name? That way rapid pushes to e.g. `main` wouldn't kick off multiple builds/deploys.

Comment (robhudson) [reply .github/workflows/pull_request_tests.yml:9]: Yeah that's the `|| github.ref` part. If it's not a PR it'll use that.

**Review by stevejalim — APPROVED** [review]
R+

File: .github/workflows/pull_request_tests.yml (hunk 1 above)
Comment (pmac) [reply .github/workflows/pull_request_tests.yml:9]: Ah! Nice!

**Pull request merged**